modules/local/failurecheck/main.nf

```diff
@@ -0,0 +1,86 @@
+process FAILURE_CHECK {
+    tag "failure_check"
+    label 'process_single'
+
+    input:
+    val fastp_fail
+    val kraken_fail
+    val bracken_fail
+
+    output:
+    path("failures_report.csv"), emit: failures_report
+
+    exec:
+    def processedIDs = [] as Set
+
+    task.workDir.resolve("failures_report.csv").withWriter { writer ->
+
+        // write header to the file
+        writer.writeLine("sample,module,error_message")
+
+        // Process FASTP_TRIM
+        if (fastp_fail.size() > 0) {
+            fastp_fail.each {
+                def id = it[0].id
+                if (id instanceof List) {
```

Comment:
An easier way might be to add `id` to a list if it's just a single element, and handle everything in the `List` case.

```groovy
if (!(id instanceof List)) {
    id = [id]
}
// handle List case
```

Reply:
So much easier! Avoids the extra loop! Thank you!
Improved in: [836cf2a](https://github.com/phac-nml/speciesabundance/pull/20/commits/836cf2a8e07f287737c04b62681e33fa8fde6e1e)